packages/http/src/mocker/index.ts

```diff
@@ -41,20 +51,52 @@ const mock: IPrismComponents<IHttpOperation, IHttpRequest, IHttpResponse, IMockH
 
       return config;
     }),
-    chain(mockConfig => negotiateResponse(mockConfig, input, resource)),
-    chain(result => assembleResponse(result, payloadGenerator))
+    Reader.chain(mockConfig => negotiateResponse(mockConfig, input, resource)),
+    Reader.chain(result => assembleResponse(result, payloadGenerator)),
+    Reader.chain(response =>
+      withLogger(logger =>
+        pipe(
+          response,
+          Either.map(r => runCallbacks({ resource, request: input.data, response: r })(logger)),
+          Either.chain(() => response)
+        )
+      )
+    )
   );
 };
 
+function runCallbacks({
+  resource,
+  request,
+  response,
+}: {
+  resource: IHttpOperation;
+  request: IHttpRequest;
+  response: IHttpResponse;
+}) {
+  return withLogger(logger =>
+    pipe(
+      Option.fromNullable(resource.callbacks),
+      Option.alt(() => Option.some([] as IHttpCallbackOperation[])),
+      Option.map(callbacks =>
+        pipe(
+          callbacks,
+          map(callback => runCallback({ callback, request, response })(logger)())
```

Comment:
Maybe let's use the full name so we do not get confused with what's going on.
```suggestion
          Array.map(callback => runCallback({ callback, request, response })(logger)())
```

Reply:
Oh, I'm quite against doing `import * as Array from 'fp-ts...`. `Array` is an existing object in JS and mixing those names may cause a big bada boom.

Maybe `FPArray`. Or just stay with `map`? WDYT?

Reply:
Ah ok, that is a good point. Let's leave it as it is, if it causes confusion we will revisit this.